### ISSUE TITLE
Create DSL for Stylesheet

### ISSUE THREAD

**Issue by nazmulidris** [r3bl-org/r3bl_rs_utils#2]
Currently the `Stylesheet` struct has to be build imperatively. Along the lines of introducing `style!` DSL for `Style` struct, we need a `stylesheet!` macro for `Stylesheet` struct.

Here's the imperative style of code.

```rust
use crossterm::event::*;
use r3bl_rs_utils::*;

fn create_stylesheet(&mut self) -> CommonResult<Stylesheet> {
  // Turquoise:  Color::Rgb { r: 51, g: 255, b: 255 }
  // Pink:       Color::Rgb { r: 252, g: 157, b: 248 }
  // Blue:       Color::Rgb { r: 55, g: 55, b: 248 }
  // Faded blue: Color::Rgb { r: 85, g: 85, b: 255 }
  throws_with_return!({
    let mut stylesheet = Stylesheet::new();

    stylesheet.add_styles(vec![
      style! {
        id: style1
        margin: 1
        color_bg: Color::Rgb { r: 55, g: 55, b: 248 }
      },
      style! {
        id: style2
        margin: 1
        color_bg: Color::Rgb { r: 85, g: 85, b: 255 }
      },
    ])?;

    stylesheet
  })
}
```

So the declarative version might look like.

```rust
use crossterm::event::*;
use r3bl_rs_utils::*;

fn create_stylesheet(&mut self) -> CommonResult<Stylesheet> {
  // Turquoise:  Color::Rgb { r: 51, g: 255, b: 255 }
  // Pink:       Color::Rgb { r: 252, g: 157, b: 248 }
  // Blue:       Color::Rgb { r: 55, g: 55, b: 248 }
  // Faded blue: Color::Rgb { r: 85, g: 85, b: 255 }
  throws_with_return!({
    stylesheet!(
        style! {
          id: style1
          margin: 1
          color_bg: Color::Rgb { r: 55, g: 55, b: 248 }
        },
        style! {
          id: style2
          margin: 1
          color_bg: Color::Rgb { r: 85, g: 85, b: 255 }
        }
    )
  })
}
```
Related:
- See: #1 
- Source code: https://github.com/r3bl-org/r3bl-rs-utils/blob/main/core/src/tui_core/styles/stylesheet.rs
- Test: https://github.com/r3bl-org/r3bl-rs-utils/blob/main/tests/test_style.rs#L158

**Comment by nazmulidris** [comment]
Fixed by c6834c17f5393728392fce28c189439cc69708d9

**Comment by nazmulidris** [comment]
This commit ad58be433eaa0c14f0ad232a838917b802f64ef7 rewrites the stylesheet. The previous implementation of the macro & the trait were "backwards"
